src/api.js

```diff
@@ -693,7 +695,7 @@ api.declare({
     let def   = await task.definition();
 
     // Compare the two task definitions
-    if (!_.isEqual(taskDef, def)) {
+    if (taskDef.hasOwnProperty('clientId') === def.hasOwnProperty('clientId') && !_.isEqual(taskDef, def)) {
```

Comment:
If the clientIds were different, I think this would just skip comparing the rest of the data.  We don't want that!

It does make sense to verify that the clientIds match here.  Perhaps the right thing to do is to set `taskDef.clientId = clientId` earlier in this function (after the check I suggested above).  Then both `taskDef` and `def` will always have a clientId, and the `_.isEqual` will be enough.